src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -186,6 +187,21 @@ void SensorFusion::updateWorld(
             }
         }
         friendly_team.setUnavailableRobotCapabilities(robot_id, unavailableCapabilities);
+
+        if (robot_status_msg.has_break_beam_status() &&
+            robot_status_msg.break_beam_status().ball_in_beam())
+        {
+            friendly_robot_id_with_ball_in_dribbler = robot_id;
+            ball_in_dribbler_timeout =
+                sensor_fusion_config->getNumDroppedDetectionsBeforeBallNotInDribbler()
+                    ->value();
+        }
+        else if (friendly_robot_id_with_ball_in_dribbler.has_value() &&
+                 friendly_robot_id_with_ball_in_dribbler.value() == robot_id)
+        {
+            friendly_robot_id_with_ball_in_dribbler = std::nullopt;
+            ball_in_dribbler_timeout                = 0;
+        }
```

Comment:
What happens if another robot prior to this `robot_status_msg` had the ball in its dribbler. Wouldnt this overwrite the `friendly_robot_id_with_ball_in_dribbler` set by that robot?

Reply:
yeah it would, hopefully the robot status messages are in order of oldest to newest?